### PR TITLE
coprocessor:init tipb && trait for dag

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,all,
       This PR prepare environment for `dag` development:
        1. update `tipb` to version include `dag`.
        2. init trait  `Executor`

@BusyJay @hhkbp2 PTAL

### CODE REVIEW

**Comment by andelf** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @hhkbp2

File: src/server/coprocessor/executor/mod.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use util::codec::table::RowColsDict;
+use server::coprocessor::Result;
+
+pub trait Executor {
+    fn next(&mut self) -> Result<Option<(i64, RowColsDict)>>;
```
Comment (siddontang) [inline src/server/coprocessor/executor/mod.rs]: better to define a structure for `(i64, RowColsDict)`

**Comment by AndreMouche** [comment]
PTAL

File: src/server/coprocessor/executor/mod.rs [hunk 2]
```diff
@@ -14,6 +14,30 @@
 use util::codec::table::RowColsDict;
 use server::coprocessor::Result;
 
+#[allow(dead_code)] //TODO:remove it
+pub struct Row {
+    handler: i64,
```
Comment (siddontang) [inline src/server/coprocessor/executor/mod.rs]: maybe handle, not handler?

File: src/server/coprocessor/executor/mod.rs [hunk 3]
```diff
@@ -14,6 +14,30 @@
...
+#[allow(dead_code)] //TODO:remove it
+impl Row {
+    pub fn new(handler: i64, data: RowColsDict) -> Row {
+        Row {
+            handler: handler,
+            data: data,
+        }
+    }
+
+    pub fn get_mut_data(&mut self) -> (&mut RowColsDict) {
```
Comment (siddontang) [inline src/server/coprocessor/executor/mod.rs]: Need to return a tuple here? 
why not implement as_mut or as_ref trait?

File: src/server/coprocessor/executor/mod.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        &mut self.data
+    }
+
+    pub fn get_data(self) -> RowColsDict {
+        self.data
```
Comment (siddontang) [inline src/server/coprocessor/executor/mod.rs]: why move here?

File: src/server/coprocessor/executor/mod.rs (hunk 3 above)
Comment (AndreMouche) [inline src/server/coprocessor/executor/mod.rs]: The `handle`  once created, it cannot be modified. It seems `RowColsDict` is enough @siddontang

**Comment by AndreMouche** [comment]
PTAL

**Comment by hhkbp2** [comment]
LGTM

**Comment by AndreMouche** [comment]
@siddontang PTAL again?

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
CI failed @AndreMouche 

```
thread 'util::threadpool::test::test_for_tasks_with_different_cost' panicked at 'assertion failed: `(left != right)` (left: `1001`, right: `1001`)', src/util/threadpool.rs:476
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: tikv::util::panic_hook::track_hook::{{closure}}
test util::codec::mysql::time::test::test_to_dec ... ok
   1: <std::thread::local::LocalKey<T>>::with
   2: tikv::util::panic_hook::track_hook
   3: core::ops::Fn::call
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::begin_panic
   6: std::panicking::begin_panic_fmt
   7: tikv::util::threadpool::test::test_for_tasks_with_different_cost
   8: <F as test::FnBox<T>>::call_box
   9: std::panicking::try::do_call
  10: __rust_maybe_catch_panic
  11: std::panicking::try::do_call
  12: __rust_maybe_catch_panic
  13: <F as alloc::boxed::FnBox<A>>::call_box
  14: std::sys::imp::thread::Thread::new::thread_start
  15: _pthread_body
  16: _pthread_start
test util::threadpool::test::test_for_tasks_with_different_cost ... FAILED
```

**Comment by AndreMouche** [comment]
I will fix it in another PR @siddontang

**Pull request merged**